doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+
+* **Handling Ambiguity:** If a reference could resolve to multiple declarations within the same scope, or if no resolution is found after checking all scopes, documentation tools should issue a warning.
+
+### **5.2. Scope Precedence Hierarchy**
+
+The resolution process for a reference `[name]` follows the standard Dart scope of the documented element with the extension of the doc imported scope at the end. Search is done in a specific order of precedence from the narrowest (most local) scope to the broadest (globally available).
+
+The hierarchy is searched from the inside out. Below is an example for an instance method:
+```
++--------------------------------------------------------------------------+
+| 7. Doc Imported Scope (documentation-only imports with @docImport )      |
+| +----------------------------------------------------------------------+ |
+| | 6. Imported Scopes (all 'import' directives + implicit 'dart:core')  | |
+| | +------------------------------------------------------------------+ | |
+| | | 5. Library Scope (all other declarations in the file)            | | |
+| | | +--------------------------------------------------------------+ | | |
+| | | | 4. Class Type Parameter Scope (e.g., <T>)                    | | | |
+| | | | +----------------------------------------------------------+ | | | |
+| | | | | 3. Class Member Scope (e.g., static/instance members)    | | | | |
+| | | | | +------------------------------------------------------+ | | | | |
+| | | | | | 2. Method Type Parameter Scope (e.g., <R>)           | | | | | |
+| | | | | | +--------------------------------------------------+ | | | | | |
+| | | | | | | 1. Formal Parameter Scope (e.g., parameter names)| | | | | | |
+| | | | | | +--------------------------------------------------+ | | | | | |
+| | | | | +------------------------------------------------------+ | | | | |
+| | | | +----------------------------------------------------------+ | | | |
+| | | +--------------------------------------------------------------+ | | |
+| | +------------------------------------------------------------------+ | |
+| +----------------------------------------------------------------------+ |
++--------------------------------------------------------------------------+
+
+```
+### **5.3. Detailed Lookup Process**
+
+The lookup process begins at a specific "starting scope"  determined by the context of the doc comment and then follows the scope hierarchy.
+
+The following sections detail the starting scope for each type of declaration.
+
+#### **5.3.1. Instance Methods and Constructors**
+
+This applies to doc comments on methods, constructors, and operators within a class, enum, mixin, extension, or extension type. The search begins with the method's own parameters.
```

Comment:
(Not setters? They have a parameter list too, but I can see that's inconsistent with treating them as properties.)